include/nbl/ext/FFT/FFT.h

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_EXT_FFT_INCLUDED_
+#define _NBL_EXT_FFT_INCLUDED_
+
+#include "nabla.h"
+#include "nbl/video/IGPUShader.h"
+#include "nbl/asset/ICPUShader.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace FFT
+{
+
+class FFT : public core::TotalInterface
+{
+	public:
+
+		enum class Direction : uint32_t {
+			X = 0,
+			Y = 1,
+			Z = 2,
+		};
+		
+		enum class PaddingType : uint32_t {
+			CLAMP_TO_EDGE = 0,
+			FILL_WITH_ZERO = 1,
+		};
+
+		enum class DataType {
+			SSBO,
+			TEXTURE2D,
+		};
+
+		struct DispatchInfo_t
+		{
+			uint32_t workGroupDims[3];
+			uint32_t workGroupCount[3];
+		};
+
+		struct alignas(16) Uniforms_t 
+		{
+			uint32_t dims[3];
+		};
+
+		// returns dispatch size and fills the uniform data
+		static inline DispatchInfo_t buildParameters(
+			asset::VkExtent3D const & paddedInputDimensions,
+			Direction direction,
+			uint32_t num_channels)
+		{
+			assert(num_channels > 0);
+			assert(core::isPoT(paddedInputDimensions.width) && core::isPoT(paddedInputDimensions.height) && core::isPoT(paddedInputDimensions.depth));
```

Comment:
move the dimension padding function to `FFT` and pad the dimension for the user!

Reply:
For example the pushConstant(helper) for FFT has 2 distinct parameters named "inputDimension" and "paddedInputDimension"

I cannot simply get the paddedDimension from inputDimension  because kernelSize might differ because paddedDimension is not just nextPOT it's also considering the kernel's dimensions.

Just like in the shader I believe padding needs to be done by the example/user.

What do you think? maybe there is a better way that I'm missing.

Reply:
@devshgraphicsprogramming

Reply:
> I cannot simply get the paddedDimension from inputDimension because kernelSize might differ because paddedDimension is not just nextPOT it's also considering the kernel's dimensions.

Provide a `static` function for computing the padded input dimension from input dimension, and make this function take a parameter `uint32_t workgroupSize` (assume workgroup is 1D) and use the static.

> Addding parameters like "sourceImageSize", "kernelImageSize", "minSize" in FFTClass would be out of scope of an extension that has the job only of computing FFT.

did I tell you to add these? are you refering to the idea of `nbl_glsl_ext_FFT_Parameters_t` ?